server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -912,11 +943,44 @@ public void test_all_eq_on_array_ref() throws Exception {
         builder.indexValue("a", listOfOneAndNull);
         builder.indexValue("a", listOfTwoAndNull);
         try (QueryTester tester = builder.build()) {
+            assertThat(tester.toQuery("1 = all(a)"))
+                .hasToString("((+NumNullTermsPerDoc: a +(+*:* -((a:[2 TO 2147483647] a:[-2147483648 TO 0])~1))) _array_length_a:[0 TO 0])~1");
             assertThat(tester.runQuery("a", "1 = all(a)"))
                 .containsExactly(List.of(1), List.of(1, 1), List.of());
         }
     }
 
+    @Test
+    public void test_all_eq_on_string_array_ref() throws Exception {
+        var listOfNulls = new ArrayList<String>();
+        listOfNulls.add(null);
+        var listOfAAndNull = new ArrayList<String>();
+        listOfAAndNull.add("a");
+        listOfAAndNull.add(null);
+        var listOfBAndNull = new ArrayList<String>();
+        listOfBAndNull.add("b");
+        listOfBAndNull.add(null);
+
+        QueryTester.Builder builder = new QueryTester.Builder(
+            THREAD_POOL,
+            clusterService,
+            Version.CURRENT,
+            "create table tbl (a string[])");
```

Comment:
Could we maybe use DataTypeTesting for randomType+randomValues to ensure this works across all types that are supported?

Reply:
Thanks, I've tried to add random types/values to test all types, but there are a few issues - can only test execution of the Lucene queries because the query generations are slightly different per type AND having trouble with geo_shape, geo_point, and object type's random value generations that can be used as inputs to all-eq operator.